src/mjolnir/graphbuilder.cc

```diff
@@ -1449,7 +1458,12 @@ void GraphBuilder::Build(const boost::property_tree::ptree& pt,
   unsigned int threads =
       std::max(static_cast<unsigned int>(1),
                pt.get<unsigned int>("mjolnir.concurrency", std::thread::hardware_concurrency()));
-  std::string tile_dir = pt.get<std::string>("mjolnir.tile_dir");
+
+  auto tile_dir = pt.get<std::string>("mjolnir.tile_dir");
+  // Disable sqlite3 internal memory tracking (results in a high-contention mutex, and we don't care
+  // about marginal sqlite memory usage).
+  sqlite3_config(SQLITE_CONFIG_MEMSTATUS, false);
```

Comment:
i never noticed this option before, how much of an impact would you say it had?

Reply:
At higher levels of concurrency (> 8), sys time dominates once the `covered_by` performance is fixed.

High-level root cause:
- The `ST_AsText(geom)` from the admin queries generates a *lot* of text
- Building the return values results in a lot of memory allocations while fetching/building the polygons' WKT.
- [SQLite3 tracks memory stats](https://www.sqlite.org/malloc.html#_memory_status) at a global level (behind a single `static` mutex).
- The mutex spinlock becomes the bottleneck around returning the admin queries (i.e., a lot of system time, and more than 50% of the profile samples are in that spinlock).

We could also fix this long-term by adjusting the approach in any of the following ways:
- Directly using SQLite for the PiP lookup. I think this might be faster because we don't have to fetch and parse for every admin poly for every tile, and then run the PiP for every node.
- Switching from `boost::geometry` to `geos` to allow us to return WKB for less (de-)serialization cost.